Dynamic/Types/RecaptchaType.php

```diff
@@ -35,13 +49,25 @@ public function getConfiguration(): FormFieldTypeConfiguration
     public function build(FormBuilderInterface $builder, FormField $field, string $locale, array $options): void
     {
         // Use in this way the recaptcha bundle could maybe not exists.
+        $constraint = new \EWZ\Bundle\RecaptchaBundle\Validator\Constraints\IsTrue();
+        $type = \EWZ\Bundle\RecaptchaBundle\Form\Type\EWZRecaptchaType::class;
         $options['mapped'] = false;
-        $options['constraints'] = new \EWZ\Bundle\RecaptchaBundle\Validator\Constraints\IsTrue();
         $options['attr']['options'] = [
             'theme' => 'light',
             'type' => 'image',
             'size' => 'normal',
         ];
-        $builder->add($field->getKey(), \EWZ\Bundle\RecaptchaBundle\Form\Type\EWZRecaptchaType::class, $options);
+
+        if ($this->params->has('ewz_recaptcha.version')
+            && 3 == $this->params->get('ewz_recaptcha.version')
+        ) {
+            $constraint = new \EWZ\Bundle\RecaptchaBundle\Validator\Constraints\IsTrueV3();
+            $type = \EWZ\Bundle\RecaptchaBundle\Form\Type\EWZRecaptchaV3Type::class;
+            unset($options['attr']);
+        }
```

Comment:
@Quehnie This should be done in a compilerpass. There you have access to the `ewz_recpatcha.version` even the bundle is registered after the form bundle: https://github.com/sulu/SuluFormBundle/blob/b01747b887e66a8dcfdfbee4332e5e329472f882/SuluFormBundle.php#L26

And then you can set a `sulu_form.recaptcha_version` parameter which you use in type_reacptacha.xml